From 7c492fa2fb222e01f10254455efc888c153d569a Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 20 Sep 2024 18:06:38 +0200
Subject: [PATCH 1250/1695] dts: stm32: add SCMI autonomous clock management
 for stm32mp25x

Adds CK_SCMI_GPIOZ_AM, CK_SCMI_LPDMA_AM, CK_SCMI_RTC_AM and
CK_SCMI_IPCC2_AM SCMI clocks.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ed1802173b3499c7c82aeb0ef87db12d48d6b583 dts: stm32: add SCMI autonomous clock management for stm32mp25x]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I69c3d9f05256c46235166edc4ed2e91ec4a39a2a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/406048
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 56 +++++++++++++++++++++++++++++++
 1 file changed, 56 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 90e2f84a5..b5e0bbcb1 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -1066,6 +1066,34 @@
 						clocks = <&rcc MSI_KER_CK>;
 						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
 					};
+
+					clock@80 {
+						reg = <CK_SCMI_GPIOZ_AM>;
+						domain-name = "ck_gpioz_am";
+						clocks = <&rcc CK_GPIOZ_AM>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@81 {
+						reg = <CK_SCMI_LPDMA_AM>;
+						domain-name = "ck_lpdma_am";
+						clocks = <&rcc CK_LPDMA_AM>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@82 {
+						reg = <CK_SCMI_RTC_AM>;
+						domain-name = "ck_rtc_am";
+						clocks = <&rcc CK_RTC_AM>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@83 {
+						reg = <CK_SCMI_IPCC2_AM>;
+						domain-name = "ck_ipcc2_am";
+						clocks = <&rcc CK_IPCC2_AM>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 				};
 			};
 
@@ -2079,6 +2107,34 @@
 						clocks = <&rcc MSI_KER_CK>;
 						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
 					};
+
+					clock@80 {
+						reg = <CK_SCMI_GPIOZ_AM>;
+						domain-name = "ck_gpioz_am";
+						clocks = <&rcc CK_GPIOZ_AM>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@81 {
+						reg = <CK_SCMI_LPDMA_AM>;
+						domain-name = "ck_lpdma_am";
+						clocks = <&rcc CK_LPDMA_AM>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@82 {
+						reg = <CK_SCMI_RTC_AM>;
+						domain-name = "ck_rtc_am";
+						clocks = <&rcc CK_RTC_AM>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@83 {
+						reg = <CK_SCMI_IPCC2_AM>;
+						domain-name = "ck_ipcc2_am";
+						clocks = <&rcc CK_IPCC2_AM>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 				};
 			};
 
-- 
2.47.2

